### PR TITLE
New authorization in SearcherEntry

### PR DESCRIPTION
**Pull request by balcirakpeter**
- In SearcherEntry was completely replaced the old authorization.
- For that purpose was updated also perun-roles.yml file with the
  policies used in the SearcherEntry.

### CODE REVIEW

File: perun-base/src/test/resources/perun-roles.yml [hunk 1]
```diff
@@ -124,4 +124,56 @@ perun_policies:
       - PERUNOBSERVER:
     include_policies:
       - test_resource_admin
+
+  #SearcherEntry
+  getUsers_Map<String, String>_policy:
```
Comment (stavamichal) [inline perun-base/src/test/resources/perun-roles.yml]: Why is this policy different from the others? There is ", " between strings in the Map.

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java [hunk 2]
```diff
@@ -53,10 +53,8 @@ public SearcherEntry() {
 	@Override
 	public List<User> getUsers(PerunSession sess, Map<String, String> attributesWithSearchingValues) throws InternalErrorException, AttributeNotExistsException, PrivilegeException, WrongAttributeAssignmentException {
 		// Authorization
-		if (!AuthzResolver.isAuthorized(sess, Role.PERUNADMIN) &&
-				!AuthzResolver.isAuthorized(sess, Role.PERUNOBSERVER)) {
+		if (!AuthzResolver.authorizedInternal(sess, "getUsers_Map<String, String>_policy", Collections.emptyList()))
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java]: I would replace ", " in the policy name for "_" like in the other cases.

File: perun-base/src/test/resources/perun-roles.yml [hunk 3]
```diff
@@ -124,4 +124,56 @@ perun_policies:
...
+    include_policies:
+      - default_policy
+
+  getMembersByExpiration_String_LocalDate_policy:
+    policy_roles:
+      - PERUNOBSERVER:
+    include_policies:
+      - default_policy
+
+  getFacilities_Map<String, String>_policy:
```
Comment (stavamichal) [inline perun-base/src/test/resources/perun-roles.yml]: The same question about ", " as above.

File: perun-base/src/test/resources/perun-roles.yml [hunk 4]
```diff
@@ -124,4 +124,56 @@ perun_policies:
...
+    policy_roles:
+      - PERUNOBSERVER:
+    include_policies:
+      - default_policy
+
+  getResources_Map<String, String>_policy:
```
Comment (stavamichal) [inline perun-base/src/test/resources/perun-roles.yml]: The same question about ", " as above.

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java [hunk 5]
```diff
@@ -161,10 +149,8 @@ public SearcherEntry() {
 	public List<Facility> getFacilities(PerunSession sess, Map<String, String> attributesWithSearchingValues) throws PrivilegeException, InternalErrorException, AttributeNotExistsException, WrongAttributeAssignmentException {
 
 		// Authorization
-		if (!AuthzResolver.isAuthorized(sess, Role.PERUNADMIN) &&
-				!AuthzResolver.isAuthorized(sess, Role.PERUNOBSERVER)) {
+		if (!AuthzResolver.authorizedInternal(sess, "getFacilities_Map<String, String>_policy", Collections.emptyList()))
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java]: The same question about ", " as above.

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java [hunk 6]
```diff
@@ -173,21 +159,17 @@ public SearcherEntry() {
 	public List<Resource> getResources(PerunSession sess, Map<String, String> attributesWithSearchingValues, boolean allowPartialMatchForString) throws PrivilegeException, InternalErrorException, AttributeNotExistsException, WrongAttributeAssignmentException {
 
 		// Authorization
-		if (!AuthzResolver.isAuthorized(sess, Role.PERUNADMIN) &&
-				!AuthzResolver.isAuthorized(sess, Role.PERUNOBSERVER)) {
+		if (!AuthzResolver.authorizedInternal(sess, "getResources_Map<String, String>_policy", Collections.emptyList()))
```
Comment (stavamichal) [inline perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java]: The same question about ", " as above.

File: perun-base/src/test/resources/perun-roles.yml (hunk 1 above)
Comment (balcirakpeter) [reply perun-base/src/test/resources/perun-roles.yml]: Fixed

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java (hunk 2 above)
Comment (balcirakpeter) [reply perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java]: Fixed

File: perun-base/src/test/resources/perun-roles.yml (hunk 3 above)
Comment (balcirakpeter) [reply perun-base/src/test/resources/perun-roles.yml]: Fixed

File: perun-base/src/test/resources/perun-roles.yml (hunk 4 above)
Comment (balcirakpeter) [reply perun-base/src/test/resources/perun-roles.yml]: Fixed

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java (hunk 5 above)
Comment (balcirakpeter) [reply perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java]: Fixed

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java (hunk 6 above)
Comment (balcirakpeter) [reply perun-core/src/main/java/cz/metacentrum/perun/core/entry/SearcherEntry.java]: Fixed

**Pull request merged**